metadata/util_test.go

```diff
@@ -46,168 +46,62 @@ import (
-				testutil.Dir("foo/bar/", testutil.WithDirOwner(1000, 1000)),
-				testutil.File("foo/bar/baz.txt", "testtest"),
-				testutil.File("foo/bar/xxxx", "x"),
-				testutil.File("foo/bar/yyy", "yyy"),
-				testutil.Dir("foo/a/", testutil.WithDirModTime(sampleTime)),
-				testutil.Dir("foo/a/1/", testutil.WithDirXattrs(map[string]string{"testkey": "testval"})),
-				testutil.File("foo/a/1/2", "1111111111"),
-			},
-			want: []check{
-				numOfNodes(9), // root dir + 4 dirs + 4 files
-				hasDirChildren("foo", "bar", "a"),
-				hasDirChildren("foo/bar", "baz.txt", "xxxx", "yyy"),
-				hasDirChildren("foo/a", "1"),
-				hasDirChildren("foo/a/1", "2"),
-				hasMode("foo", os.ModeDir|0600|os.ModeSticky),
-				hasOwner("foo/bar", 1000, 1000),
-				hasModTime("foo/a", sampleTime),
-				hasXattrs("foo/a/1", map[string]string{"testkey": "testval"}),
-				hasFile("foo/bar/baz.txt", 8),
-				hasFile("foo/bar/xxxx", 1),
-				hasFile("foo/bar/yyy", 3),
-				hasFile("foo/a/1/2", 10),
-			},
-		},
-		{
-			name: "hardlinks",
-			in: []testutil.TarEntry{
-				testutil.File("foo", "foofoo", testutil.WithFileOwner(1000, 1000)),
-				testutil.Dir("bar/"),
-				testutil.Link("bar/foolink", "foo"),
-				testutil.Link("bar/foolink2", "bar/foolink"),
-				testutil.Dir("bar/1/"),
-				testutil.File("bar/1/baz.txt", "testtest"),
-				testutil.Link("barlink", "bar/1/baz.txt"),
-				testutil.Symlink("foosym", "bar/foolink2"),
-			},
-			want: []check{
-				numOfNodes(6), // root dir + 2 dirs + 1 flie(linked) + 1 file(linked) + 1 symlink
-				hasFile("foo", 6),
-				hasOwner("foo", 1000, 1000),
-				hasFile("bar/foolink", 6),
-				hasOwner("bar/foolink", 1000, 1000),
-				hasFile("bar/foolink2", 6),
-				hasOwner("bar/foolink2", 1000, 1000),
-				hasFile("bar/1/baz.txt", 8),
-				hasFile("barlink", 8),
-				hasDirChildren("bar", "foolink", "foolink2", "1"),
-				hasDirChildren("bar/1", "baz.txt"),
-				sameNodes("foo", "bar/foolink", "bar/foolink2"),
-				sameNodes("bar/1/baz.txt", "barlink"),
-				linkName("foosym", "bar/foolink2"),
-				hasNumLink("foo", 3),     // parent dir + 2 links
-				hasNumLink("barlink", 2), // parent dir + 1 link
-				hasNumLink("bar", 3),     // parent + "." + child's ".."
-			},
-		},
-		{
-			name: "various files",
-			in: []testutil.TarEntry{
-				testutil.Dir("bar/"),
-				testutil.File("bar/../bar///////////////////foo", ""),
-				testutil.Chardev("bar/cdev", 10, 11),
-				testutil.Blockdev("bar/bdev", 100, 101),
-				testutil.Fifo("bar/fifo"),
-			},
-			want: []check{
-				numOfNodes(6), // root dir + 1 file + 1 dir + 1 cdev + 1 bdev + 1 fifo
-				hasFile("bar/foo", 0),
-				hasChardev("bar/cdev", 10, 11),
-				hasBlockdev("bar/bdev", 100, 101),
-				hasFifo("bar/fifo"),
-			},
-		},
+// newTestableReader creates a test bbolt db as-well as a metadata.Reader given a TOC.
+func newTestableReader(sr *io.SectionReader, toc ztoc.TOC, opts ...Option) (testableReader, error) {
+	db, clean, err := newTempDB("")
```

Comment:
Will this just create the temp db at root? Maybe we should allow it to be configurable, e.g. make a tempdir and pass it into the TestableReader or something.

Reply:
If no dir is provided (""), It will create the temp DB under`/tmp`. If you provide a dir it will create the temp DB file under that dir.